### PR TITLE
spelling corrections etc

### PR DESCRIPTION
**Pull request by PublicHealthDataGeek**
some spelling corrections, some comments for RL to consider, some definitions added for clarity, some comments added to code.

### CODE REVIEW

File: 02-basics.Rmd [hunk 1]
```diff
@@ -7,9 +7,9 @@ This chapter may seem tedious for people who just want to crack on and load-in d
 However, working through the examples below is recommended for most people unless you're already an experienced R user, and even experienced R users may learn something about the language's unique syntax in the following sections.
 
 The first step is to **start RStudio**, e.g. by tapping the Start button and searching for RStudio if you are on Windows.
-You should see an 'R console' like that displayed in Figure \@ref(fig:console).
+You should see an 'R console pane' like that displayed in Figure \@ref(fig:console).
 
-```{r console, echo=FALSE, fig.cap='The R console in RStudio. You should see something like this when you first open RStudio. Tapping Enter at this point will trigger the autocompletion of the command `install.packages("stats19").'}
+```{r console, echo=FALSE, fig.cap='The R console pane in RStudio. You should see something like this when you first open RStudio. Having entered the text 'install.packages("stats', tapping 'Enter' at this point will trigger the autocompletion of the command `install.packages("stats19").'}
```
Comment (Robinlovelace) [inline 02-basics.Rmd:12]: This is definitely much better. 'Tapping' is a funny word, would 'typing Enter' be better?... Probably not.

File: 02-basics.Rmd [hunk 2]
```diff
@@ -20,7 +20,7 @@ You are ready to start running R code, by entering commands into the console.
 
 R can be understood as a giant calculator.
 If you feed the console arithmetic tasks, it will solve them precisely and instantly.
-Try typing the following for examples (note that `pi` is an inbuilt object) into the R console in RStudio and pressing `Enter` to make R run the code:
+Try typing the following examples (note that `pi` is an inbuilt object) into the R console in RStudio and pressing `Enter` to make R run the code.  The output of the code, when shown in this book, is preceeded by '##'.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:23]: That is a very good point you make about `##`.

File: 02-basics.Rmd [hunk 3]
```diff
@@ -36,10 +36,10 @@ sqrt(361)
 This is all well and good, providing a chance to see how R works with numbers and get practice with typing commands into the console.
 However, a key benefit of R is that it is *object oriented*, meaning it stores values and complex objects such as data frames representing road casualties and processes them *in memory* (meaning that R is both fast and memory hungry when working with large datasets).
 
-The two most common ways of creating objects are with the the `<-` 'arrow' or `=` 'equals' assignment operators.^[
+The two most common ways of creating objects are using `<-` 'arrow' or `=` 'equals'.  These are known as assignment operators as they assign contents(?), such as numbers, to objects.^[
```
Comment (Robinlovelace) [inline 02-basics.Rmd:39]: Thanks for this, definite improvement in terms of readability. 'they assign contents' is fine.

File: 02-basics.Rmd [hunk 4]
```diff
@@ -48,13 +48,13 @@ z = x + y
 pi^z / exp(z)
 ```
 
-The previous code chunk created three objects called `x`, `y` and `z` and showed how these objects can themselves be used to create additional objects.
+The previous code chunk created and stored three objects called `x`, `y` and `z` and showed how these objects can themselves be used to create additional objects.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:51]: :+1:

File: 02-basics.Rmd [hunk 5]
```diff
@@ -65,9 +65,9 @@ rm(x, y, z)
 ```
 
 What just happened?
-We have removed the objects using the R function `rm()`, which stands for 'remove'.
+We have removed the objects using the R function `rm()`, which stands for 'remove'. A function is an instruction or set of instructions for R to do something with what we give to that function.  What we give to the function are known as arguments.  Each function has set of arguments we can potentially give to it.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:68]: Important clarification.

File: 02-basics.Rmd [hunk 6]
```diff
@@ -65,9 +65,9 @@ rm(x, y, z)
 ```
 
 What just happened?
-We have removed the objects using the R function `rm()`, which stands for 'remove'.
+We have removed the objects using the R function `rm()`, which stands for 'remove'. A function is an instruction or set of instructions for R to do something with what we give to that function.  What we give to the function are known as arguments.  Each function has set of arguments we can potentially give to it.  
 Technically speaking, we *passed the objects to arguments in the `rm()` function call*.
-In plain English, things that go inside the curved brackets that follow a function name are arguments.
+In plain English, things that go inside the curved brackets that follow a function name are the arguments.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:70]: How about:

```
In plain English, things that go inside the curved brackets following a function name are the function's *arguments*.
```

Even more specific.

File: 02-basics.Rmd [hunk 7]
```diff
@@ -132,13 +132,13 @@ ncol(crashes)
 The results of the previous commands tell us that the dataset has 3 rows and 2 columns.
 We will use larger datasets, with thousands of rows and tens of columns, in later chapters but for now it's good to 'start small' to understand the basics of R.
 
-## Subsetting by index or name
+## Subsetting by index or name CHANGE TITLE OR BE CLEARER WHEN YOU ARE SUBSETTING BY INDEX OR NAME IN THE TEXT BELOW
```
Comment (Robinlovelace) [inline 02-basics.Rmd]: Good point. How about just 'Subsetting'? 

Would

'Subsetting vectors' 

be any better?

File: 02-basics.Rmd [hunk 8]
```diff
@@ -132,13 +132,13 @@ ncol(crashes)
...
 
 Subsetting returns part of an R object. 
-It can be done by providing numbers representing the positions of the elements we want (e.g. the 2^nd^ element) or with a logical vector, with values associated with `TRUE` returned. 
+It can be done by providing numbers representing the positions of the elements we want (e.g. the 2^nd^ element) or with a logical vector *NEED TO EXPLAIN VECTOR, with values associated with `TRUE` or `FALSE` returned.
```
Comment (Robinlovelace) [inline 02-basics.Rmd]: Good point. How about:

```
The most basic type of R object is a *vector*, a sequence of values of the same type.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) or length 3; `casualty_age` is a *numeric vector* of length three.
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected. 
```

File: 02-basics.Rmd [hunk 9]
```diff
@@ -214,7 +214,7 @@ crashes$casualty_age[crashes$vehicle_type == "tank"] = 61
 
 ## Dealing with NAs and recoding
 
-R objects can have a value of NA. This is how R represents missing data.
+R objects can have a value of NA. NA is how R represents missing data.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:220]: :+1:

File: 02-basics.Rmd [hunk 10]
```diff
@@ -260,7 +260,7 @@ as.matrix(crashes)
 
 Often it is useful to 'recode' values.
 In the raw STATS19 files, for example, -1 means NA.
-There are many ways to recode values in R, the simplest and most mature of which is the use of factors, as shown below:
+There are many ways to recode values in R, the simplest and most mature of which is the use of factors DEFINE WHAT A FACTOR IS, as shown below:
```
Comment (Robinlovelace) [inline 02-basics.Rmd]: How about:

```
 There are many ways to recode values in R, the simplest and most mature of which is the use of factors, which are whole numbers representing characters, as shown below: 
```

On second thoughts, is this even a useful example? It is a bit confusing I think so it may be worth scrapping it IMO...

File: 02-basics.Rmd [hunk 11]
```diff
@@ -275,7 +275,7 @@ z_charcter
 
 ## Saving R objects
 
-You can also save individual R objects in the RDS format.
+You can also save individual R objects in the RDS format. The RDS format is the ? data format for R. (? is this the right term?)
```
Comment (Robinlovelace) [inline 02-basics.Rmd]: Yes that's the right term. How about the following?

``` 
You can also save individual R objects in the Rds format. The Rds format is R's native data format, meaning that any R object can be saved as an Rds file.
```

File: 02-basics.Rmd [hunk 12]
```diff
@@ -291,7 +291,7 @@ identical(crashes, crashes2)
 R also supports many other formats, including CSV files, which can be created and imported with the functions `readr::read_csv()` and `readr::write_csv()` (see also the [readr](https://readr.tidyverse.org/) package).
 
 ```{r readr-write, eval=FALSE}
-readr::write_csv(crashes, "crashes.csv")
+readr::write_csv(crashes, "crashes.csv") # This code tells R to use the write_csv function from the readr package
```
Comment (Robinlovelace) [inline 02-basics.Rmd:297]: :+1:

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Great set of suggestions. I've suggested a few tweaks before merging. Thanks Caroline!

File: 02-basics.Rmd (hunk 1 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:12]: Yes it would

File: 02-basics.Rmd (hunk 2 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:23]: not sure if I need to comment or not to this...Just my lack of knowledge on how to use github again!

File: 02-basics.Rmd (hunk 3 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:39]: :+1:

File: 02-basics.Rmd (hunk 5 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:68]: :)

File: 02-basics.Rmd (hunk 4 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:51]: :)

File: 02-basics.Rmd (hunk 6 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:70]: Yes - even better!

File: 02-basics.Rmd (hunk 7 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Would need to define vector which I mentioned below. If 2.3 was subsetting then would you merge it with 2.4 (which is also subsetting)?  I think you could keep as subsetting by index or name if you make it clearer in the text what 'index' and 'name' mean.

Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: ah looks like you have sorted it below!!!

File: 02-basics.Rmd (hunk 8 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Few more edits below- hope they make sense 


The most basic type of R object is a *vector*, a sequence of values of the same type, FOR EXAMPLE, TYPE COULD BE NUMERIC.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3. CHANGED THREE TO 3 FOR COSNISTENCY
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want WITHIN THE VECTOR (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

File: 02-basics.Rmd (hunk 10 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: There are quite a few use cases for using factors in STATS19 eg recoding vehicles to simpler categories so I think it is helpful to have a look at factors.  I think the wording you have suggested is better - perhaps worth saying something that factors are usually used for categorical data - i think that would help - would a male/femae/other example would be more helpful?

File: 02-basics.Rmd (hunk 11 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Perhaps you could reference it to Excel syaing that 'in the same way that .xls is the native Excel data format.  ?

Comment (Robinlovelace) [reply 02-basics.Rmd]: good idea

File: 02-basics.Rmd (hunk 2 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd:23]: :)

**Comment by Robinlovelace** [comment]
Thanks for all the replies Caroline, it seems we're on the same wavelength. Do you want to have one more go at changing a few things? If so let me know when you're done (you can edit this version, let me know if you have any issues finding out how) and I'll merge this, can always tidy-up after :broom:

File: 02-basics.Rmd (hunk 8 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: SUGGESTED WORDING CHANGES BELOW

The most basic type of R object is a *vector*, a sequence of values of the same type FOR EXAMPLE, A SET OF NUMBER.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3(CHANGED TO THREE FOR CONSISTENCY).


In R, subsets of vectors can be returned by providing numbers representing the positions of the elements IN THE VECTOR  we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

File: 02-basics.Rmd (hunk 10 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: I think the wording is now good.  A better example might be to use gender e.g male, female, other as that is more relevent and highlights how we tend to use factors - to code categorical data.  Perhaps add a sentence saying that are usually used for categorical variables.

File: 02-basics.Rmd (hunk 11 above)
Comment (PublicHealthDataGeek) [reply 02-basics.Rmd]: Suggest add the wording below or similar:  
This is comparable to Excel where the native data format is xls.

**Comment by PublicHealthDataGeek** [comment]
Hi Robin, thanks for the feedback.  I dont know what quite happened - I put in various replies then looked at help and thought that I had lost all my comments but seeing your reply 5 mins ago it looks like you might have got them...?  anyway, I had gone through again and done them again and so you might get them again.  I am not sure how to 'changes things and edit this version'?  Do I just refresh the book and then go through and edit again?

**Comment by Robinlovelace** [comment]
Hi @PublicHealthDataGeek I can see all of your comments but some of them show as 'resolved' if you edit sections as shown below, nothing has been lost.

image

**Comment by Robinlovelace** [comment]
Can you try making a few more edits here @PublicHealthDataGeek?: https://github.com/PublicHealthDataGeek/rrsrr/edit/patch-3/02-basics.Rmd?pr=%2Fsaferactive%2Frrsrr%2Fpull%2F6

Partly for interest and partly to make the editing process quicker when I merge this PR.

**Comment by PublicHealthDataGeek** [comment]
HI Robin - used the link - it has posted it under your comment yestereday.  Then I clicked on 'request review'above so fingers crossed it all links seemlessly and you can see the futher changes I made to incorporate our discussion.....

File: 02-basics.Rmd [hunk 13]
```diff
@@ -275,7 +278,7 @@ z_charcter
 
 ## Saving R objects
 
-You can also save individual R objects in the RDS format. The RDS format is the ? data format for R. (? is this the right term?)
+You can also save individual R objects in the RDS format. The RDS format is the data format for R, meaning that any R object can be saved as an RDS file. This is equivalent to xls being the data format for Excel.
```
Comment (Robinlovelace) [inline 02-basics.Rmd:281]: Many thanks for the update @PublicHealthDataGeek !

**Comment by Robinlovelace** [comment]
Great news @PublicHealthDataGeek it works, this is what I see: 

image

**Review by Robinlovelace — APPROVED** [review]
Many thanks for the updates Caroline, way better.

**Comment by Robinlovelace** [comment]
Good practice:

image

**Comment by Robinlovelace** [comment]
Heads-up @PublicHealthDataGeek I've just cleaned up Chapter 2. Please take a look at the new shiny results here: https://saferactive.github.io/rrsrr/basics.html
